### ISSUE TITLE
Limiter isn't blocking any requests

### ISSUE THREAD

**Issue by im-kent** [express-rate-limit/express-rate-limit#481]
### Description

Below is the code I am using. When this server runs on the cloud, the rate limiting isn't working. I can make multiple post requests in quick succession to the endpoint and they will all return the empty object as expected.

Any ideas what might be going wrong? I have also tried removing `myRouter.use('/submit-responses', limiter)` and instead putting the limiter directly into the endpoint as `myRouter.post('/submit-responses', limiter, async (req, res) => {...})`, but that also didn't work in rate limiting myself. 

```
// server.js
const express = require('express');
const bodyParser = require('body-parser');
const path = require('path')
const axios = require('axios');
const dotenv = require('dotenv');
dotenv.config()

const rateLimit = require('express-rate-limit');


var myRouter = express.Router();
const PORT = 3000;

myRouter.use(bodyParser.json());

myRouter.use(express.static(path.join(__dirname, 'public')));




myRouter.get('/', (req, res) => {
    res.sendFile(path.join(__dirname, 'public', 'index.html'));
});

const limiter = rateLimit({
    windowMs: 60 * 1000, // 60 seconds
    max: 2, // Limit each IP to 2 request per windowMs
    message: {
      error: 'Rate limit exceeded. Try again in 10 seconds.',
    },
    keyGenerator: (req) => { return req.headers['x-forwarded-for'] || req.ip}, // Use IP address as the key
  });
  
myRouter.use('/submit-responses', limiter)

myRouter.post('/submit-responses', async (req, res) => {
    var userResponses = req.body; // Expecting an array of objects
    console.log('Received responses:', userResponses);
    res.send({})
    
});

module.exports = myRouter
```

### Library version

7.4.0

### Node version

20.12.2

### Module system

CommonJS

**Comment by nfriedly** [comment]
Hi @im-kent,

My first guess is that it might be something like the port number getting added to the x-forwarded-for header. Try running it without a custom `keyGenerator` and see if it logs any warnings. (The default `keyGenerator` runs a few checks for this and other common issues on the first request.) See https://express-rate-limit.mintlify.app/guides/troubleshooting-proxy-issues#port-numbers-in-ip-addresses for a bit more info about that and an example `keyGenerator` that can fix it.

However, I also see another issue in your code: accepting `req.headers['x-forwarded-for']` without any processing can allow a malicious end-user to bypass your rate limiting by setting that header to a random value on each request. (Many reverse-proxies will append to the header rather than overwriting it.) The correct way to handle this is to set `app.set('trust proxy', 1)`, which will cause express to set `req.ip` to the value from the nearest reverse-proxy, and then see if you need a higher number. (The default `keyGenerator` will also check this and log a warning if things are configured incorrectly.) See https://express-rate-limit.mintlify.app/guides/troubleshooting-proxy-issues#the-global-limiter-problem for more info about that.

**Comment by im-kent** [comment]
Thank you for the response. I have tried removing keyGenerator completely and created a new endpoint for easier testing. The new endpoint is a GET at '/ip' and this is now the endpoint where the rate limiter is now applied. This code is below.

Some notes:
1. These two have the same values: req.ip and req.socket.remoteAddress
2. The value is of the format: ::ffff:12.345.678.901
3. The IP address is not my IP address (likely as you are saying some load balancer's ip?)
4. Multiple API calls result in the IP address changing (but also some IP addresses being used again, potentially further suggesting it is some load balancer's ip?)
5. On the 3rd and 4th time the same IP showed up within the same minute, client receives message "Rate limit exceeded. Try again in 10 seconds."
6. **While (5) is encouraging to see, I expected that the IP NOT be logged on the 3rd and 4th time the same IP showed up. Since the logging was done within the GET endpoint, if the rate limiting indeed happened, I would expect that the request is blocked from hitting that endpoint completely and so no code within that endpoint should execute.** 


I would love any advice you may have.

```
// server.js
const express = require('express');
const bodyParser = require('body-parser');
const path = require('path')
const axios = require('axios');
const dotenv = require('dotenv');
dotenv.config()

const rateLimit = require('express-rate-limit');


var myRouter = express.Router();
const PORT = 3000;

myRouter.use(bodyParser.json());

myRouter.use(express.static(path.join(__dirname, 'public')));




myRouter.get('/', (req, res) => {
    res.sendFile(path.join(__dirname, 'public', 'index.html'));
});

//Change start

const limiter = rateLimit({
    windowMs: 60 * 1000, // 60 seconds
    max: 2, // Limit each IP to 1 request per windowMs
    message: {
      error: 'Rate limit exceeded. Try again in 10 seconds.',
    },
    // keyGenerator: (req) => { return req.headers['x-forwarded-for'] || req.ip}, // Use IP address as the key
  });
  
myRouter.use('/ip', limiter)

myRouter.get('/ip', async (req, res) => {
    console.log(req.ip)
    console.log(req.socket.remoteAddress)
    res.send({})
} )

//Change end


myRouter.post('/submit-responses', async (req, res) => {
    var userResponses = req.body; // Expecting an array of objects
    console.log('Received responses:', userResponses);
    res.send({})
    
});

module.exports = myRouter
```

Not sure if this is relevant at all, but I figure I'll share it just in case it is. In my NodeJS configuration, you'll notice that the express.Router() I created in the code above (myRouter) is exported. This is then consumed by my actual express server that is run in order to create a subdomain specifically for the myRouter routes.

In other words, I have a serverMain.js that I am directly running and that serverMain.js has the code below. 

```
var subdomain = require('express-subdomain');
var path = require('path');
var myRouter = require('./websites/subDomain1/myRouter')

var express = require('express');
var app = express();

app.use(subdomain('subDomain1', myRouter));

app.use(express.static(path.join(__dirname, 'websites', 'home', 'public')));
app.get('/', function(req, res) {
    res.sendFile(path.join(__dirname, 'websites', 'home', 'public', 'index.html'))
});

// Optional catch-all route for undefined subdomains
app.use(function(req, res, next) {
    res.send('404');

});

app.listen(3000);
```

**Comment by nfriedly** [comment]
Aah, that makes sense. Glad it's working for you now!
